crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -26,6 +32,134 @@ struct Metrics {
     native_price_cache_outdated_entries: IntGauge,
 }
 
+/// Trait for fetching a batch of native price estimates.
+#[mockall::automock]
+#[async_trait]
+pub trait NativePriceBatchFetcher: Sync + Send {
+    /// Maximum batch size the `fetch_native_prices()` can take
+    fn max_batch_size(&self) -> usize;
```

Comment:
Since each provider may have different maximum batch, this must be part of the trait.